src/net/slashie/serf/ui/oryxUI/ContinuousGFXUISelector.java

```diff
@@ -96,195 +88,263 @@ public void keyPressed(KeyEvent e) {
 		x2 = r.x + r.width;
 		y1 = r.y;
 		y2 = r.y + r.height;
-		
-		
+
 	}
-	
-	protected MouseMotionListener getCursorListener() {
-		return new MouseMotionListener(){
-			public void mouseDragged(MouseEvent e) {
+
+	protected MouseMotionListener getCursorListener()
+	{
+		return new MouseMotionListener()
+		{
+			public void mouseDragged(MouseEvent e)
+			{
+				// logger.debug("mouse dragged");
 				int newQuadrant = defineQuadrant(e.getPoint().x, e.getPoint().y);
-				if (mouseDirection != -1 && mouseDirection != QDIRECTIONS[newQuadrant-1]){
-					mouseDirection = QDIRECTIONS[newQuadrant-1];
+				if (mouseDirection != -1 && mouseDirection != QDIRECTIONS[newQuadrant - 1])
+				{
+					mouseDirection = QDIRECTIONS[newQuadrant - 1];
 				}
 				mouseMoved(e);
 			}
 
-			public void mouseMoved(MouseEvent e) {
+			public void mouseMoved(MouseEvent e)
+			{
+				// logger.debug("mouse moved");
 				if (!selectionActive)
 					return;
 				mousePosition = e.getPoint();
 				int newQuadrant = defineQuadrant(e.getPoint().x, e.getPoint().y);
-				si.setCursor(QCURSORS[newQuadrant-1]);
+				si.setCursor(QCURSORS[newQuadrant - 1]);
 			}
 		};
 	}
-	
 
-
-	protected MouseListener getMouseClickListener() {
-		javax.swing.Action gotoDirectionAction = new AbstractAction() {
+	protected MouseListener getMouseClickListener()
+	{
+		javax.swing.Action gotoDirectionAction = new AbstractAction()
+		{
 			private static final long serialVersionUID = 1L;
+
 			@Override
-			public void actionPerformed(ActionEvent arg0) {
+			public void actionPerformed(ActionEvent arg0)
+			{
+				// logger.debug("mouse clicked");
 				if (!selectionActive)
 					return;
 				int quadrant = defineQuadrant(mousePosition.x, mousePosition.y);
-				selectedMouseDirection = QDIRECTIONS[quadrant-1];
+				selectedMouseDirection = QDIRECTIONS[quadrant - 1];
 			}
 		};
 		final Timer gotoDirectionTimer = new Timer(200, gotoDirectionAction);
-		return new MouseAdapter() {
-			public void mousePressed(final MouseEvent e) {
+		return new MouseAdapter()
+		{
+			public void mousePressed(final MouseEvent e)
+			{
+				if (SwingUtilities.isRightMouseButton(e))

```

Comment:
Please confirm if the SwingUtilities.isRightMouseButton checks are the ones you added?